app/controllers/application_controller.rb

```diff
@@ -1,7 +1,14 @@
 class ApplicationController < ActionController::Base
-  before_action :current_user
+  protect_from_forgery with: :exception
 
-  def current_user
-    User.first
+  before_action :update_allowed_parameters, if: :devise_controller?
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :bio, :photo, :email, :password) }
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :surname, :email, :password, :current_password)
+    end
   end
 end
```

Comment:
**OPTIONAL** - You have done well so far and your devise works quite well, but I am still able to have access to the application even when I am not assigned in. A likely reason for this is that you have not added the `:authenticate_user!` method `before_action` in your application controller. Doing this will ensure only logged-in users can access your app. Please fix this 🙏 

Your project:
![xfxftnxdt](https://github.com/adamilare/railsblog/assets/73607512/2281ed57-20a1-4203-acdb-464237e5c5f6)